### ISSUE TITLE
Delegate script context['cookies'] may contain invalid cookies

### ISSUE THREAD

**Issue by kaij** [cantaloupe-project/cantaloupe#370]
Just a side note as I recently had issues with cookie reading in the delegate script. Since ruby hashes cannot contain dots in the keys, the cookies containing dots cannot be read correctly and have to be parsed manually using context['request_headers']['cookie']

Example cookie - key "_xx_yy.1.d1f7", value "ff57d4a2c3f05993.1583682973.12.1586169744.1586169744."
```
_xx_yy.1.d1f7: ff57d4a2c3f05993.1583682973.12.1586169744.1586169744.
```
will be in the hashmap as "d6f7" => "ee57d4a2c3f05993"

Suggestion: add a comment in the delegate script, add a note in the documentation, replace dots with '_' when setting the cookies.

**Comment by adolski** [comment]
Hi @kaij,

It's true that symbol-type hash keys can't contain dots, but the hash keys in the delegate script are all strings, which can. Here is what I see on the `release/4.1` branch:

```
curl -H "Cookie: _xx_yy.1.d1f7=ff57d4a2c3f05993.1583682973.12.1586169744.1586169744." http://localhost:8182/iiif/2/test.jpg/info.json
```
```
# authorize() in delegates.rb
puts context['cookies']
> {"Cookie"=>"_xx_yy.1.d1f7=ff57d4a2c3f05993.1583682973.12.1586169744.1586169744."}
```

(There is a related issue (#329) which is that the structure of this hash is wrong. It's fixed on `develop` but I didn't include the fix in the `release/4.1` branch to avoid breaking people's scripts.)

**Comment by kaij** [comment]
Hi @adolski, thanks for the analysis! Sorry, I'm actually using the current develop branch since we are using the great XMP integration. 

`context['request_headers']['Cookie']` is correct (and can be parsed in the delegate script as a workaround) , Keys in `context['cookies']` are not correct (if they contain dots).

Is it maybe the COOKIE_PATTERN which should include the dots (and maybe other characters like whitespaces for the value)? If you think this might be the place I can test and/or put together a pull request.

https://github.com/cantaloupe-project/cantaloupe/commit/e36aa3d47fc6984a0c02e0b7d4fb8f6e0e5bb465#diff-aea903254bf2257451a74a93af07a1ddR22

**Comment by adolski** [comment]
Hi @kaij,

After looking into this further, I believe it was fixed a while back, in 39459c34c11bf0abdf14ee2bd55004824ecc0db4. `Request.COOKIE_PATTERN` was removed in that commit, and I can't reproduce the problem on `develop` anymore.

(Possibly related to #360)

**Comment by kaij** [comment]
@adolski Thanks for checking! Turns out I was on a slightly older version (march 2020). I can confirm this is working as expected now on the current version.
